src/Build/Logging/TerminalLogger/ForwardingTerminalLogger.cs

```diff
@@ -0,0 +1,123 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using Microsoft.Build.Framework;
+
+namespace Microsoft.Build.Logging;
+
+/// <summary>
+/// A forwarding logger that forwards events relevant to implementing the TerminalLogger from build nodes to the central nodes.
+/// This logger
+/// </summary>
+public sealed partial class ForwardingTerminalLogger : IForwardingLogger
+{
+    /// <inheritdoc />
+    public IEventRedirector? BuildEventRedirector { get; set; }
+    /// <inheritdoc />
+    public int NodeId { get; set; }
+
+    public LoggerVerbosity Verbosity { get; set; } = LoggerVerbosity.Diagnostic;
+    /// <inheritdoc />
+    public string? Parameters { get; set; }
+
+    /// <inheritdoc />
+    public void Initialize(IEventSource eventSource, int nodeCount)
+    {
+        NodeId = nodeCount;
+        Initialize(eventSource);
+    }
+
+    /// <inheritdoc />
+    public void Initialize(IEventSource eventSource)
+    {
+        eventSource.BuildStarted += ForwardEventUnconditionally;
+        eventSource.BuildFinished += ForwardEventUnconditionally;
+        eventSource.ProjectStarted += ForwardEventUnconditionally;
+        eventSource.ProjectFinished += ForwardEventUnconditionally;
+        eventSource.TargetStarted += ForwardEventUnconditionally;
+        eventSource.TargetFinished += ScrubOutputsFromIrrelevantTargets;
+        eventSource.TaskStarted += TaskStarted;
+
+        eventSource.MessageRaised += MessageRaised;
+        eventSource.WarningRaised += ForwardEventUnconditionally;
+        eventSource.ErrorRaised += ForwardEventUnconditionally;
+        eventSource.StatusEventRaised += ForwardEvaluationEvents;
+
+        if (eventSource is IEventSource3 eventSource3)
+        {
+            eventSource3.IncludeTaskInputs();
+        }
+
+        if (eventSource is IEventSource4 eventSource4)
+        {
+            eventSource4.IncludeEvaluationPropertiesAndItems();
+        }
+    }
+
+    private void ForwardEvaluationEvents(object sender, BuildStatusEventArgs e)
+    {
+        if (e is ProjectEvaluationStartedEventArgs ||
+            e is ProjectEvaluationFinishedEventArgs)
+        {
+            // Forward evaluation events unconditionally
+            BuildEventRedirector?.ForwardEvent(e);
+        }
```

Comment:
I'm actually not sure if these even make sense - would any non-central node ever get ProjectEvaluation Start/Finish events? Isn't all evaluation on the central node?